doc/devices/gbs.rst

```diff
@@ -0,0 +1,160 @@
+The GBS device
+==============
+
+The GBS device gives access to a near-term model of photonic quantum computing called `Gaussian
+boson sampling <https://strawberryfields.ai/photonics/concepts/gbs.html>`__. This model can
+encode graphs and other quantities that can be represented by a symmetric matrix :math:`A`, with
+a number of known `applications <https://strawberryfields.ai/photonics/applications.html>`__.
+The output of a GBS device is a collection of samples listing the number of photons counted in
+each mode, resulting in a probability distribution that captures information about the encoded
+matrix.
+
+The GBS device can then be trained by varying an initial matrix according to some trainable
+parameters. When doing this, we can access the output GBS probability distribution as well as its
+derivative.
+
+Usage
+~~~~~
+
+You can instantiate the GBS device in PennyLane as follows:
+
+.. code-block:: python
+
+    import pennylane as qml
+
+    dev = qml.device('strawberryfields.gbs', wires=4, cutoff_dim=4)
+
+Note that a ``cutoff_dim`` argument is used to truncate the number of photons in the output GBS
+probability distribution.
+
+The GBS device represents a fixed circuit composed of:
+
+- Varying an input matrix according to some trainable parameters and embedding the result
+  into GBS,
+- Measuring the output probability distribution.
+
+Hence, the ``quantum_function`` used to construct the QNode must follow a restricted pattern:
+
+.. code-block:: python
+
+    from pennylane_sf.ops import ParamGraphEmbed
+
+    @qml.qnode(dev)
+    def quantum_function(x):
+        ParamGraphEmbed(x, A, n_mean, wires=range(4))
+        return qml.probs(wires=range(4))
```

Comment:
More generally, what do we expect a VGBS qnode to return? Probabilities will scale badly for devices with more wires. My first thought is to return samples and always specify `shots` when defining the device

Reply:
Returning samples is problematic for then accessing the gradient :thinking:

I think probabilities is good since you can set `analytic=False` and the number of shots, which will reconstruct the probability distribution and its derivatives from samples.

There could then be the question of whether we always make it `analytic=False`, but even though things scale badly I've had a better experience with ~4 modes using `analytic=True` than with a finite number of shots.